Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Extensions/ApiGatewayResponseExtensions.cs

```diff
@@ -0,0 +1,276 @@
+    /// <param name="statusCode">The status code to set.</param>
+    private static void SetContentTypeAndStatusCode(HttpResponse response, IDictionary<string, string>? headers, IDictionary<string, IList<string>>? multiValueHeaders, int statusCode, ApiGatewayEmulatorMode emulatorMode)
+    {
+        string? contentType = null;
+
+        if (headers != null && headers.TryGetValue("Content-Type", out var headerContentType))
+        {
+            contentType = headerContentType;
+        }
+        else if (multiValueHeaders != null && multiValueHeaders.TryGetValue("Content-Type", out var multiValueContentType))
+        {
+            contentType = multiValueContentType[0];
+        }
+
+        if (contentType != null)
+        {
+            response.ContentType = contentType;
+        }
+        else
+        {
+            if (emulatorMode == ApiGatewayEmulatorMode.HttpV1)
+            {
+                response.ContentType = "text/plain; charset=utf-8";
+            } 
+            else if (emulatorMode == ApiGatewayEmulatorMode.Rest)
+            {
+                response.ContentType = "application/json";
+            }
+        }
+
+        if (statusCode != 0)
+        {
+            response.StatusCode = statusCode;
+        }
+        else
+        {
+            if (emulatorMode == ApiGatewayEmulatorMode.Rest) // rest api text for this message is slightly different
+            {
+                response.StatusCode = 500;
+                response.ContentType = "application/json";
+                var errorBytes = Encoding.UTF8.GetBytes("{\"message\": \"Internal server error\"}");
+                response.Body = new MemoryStream(errorBytes);
+                response.ContentLength = errorBytes.Length;
+            } else
+            {
+                SetInternalServerError(response);
+            }
+        }
+    }
+
+    /// <summary>
+    /// Sets the content type and status code for API Gateway v2 responses.
+    /// </summary>
+    /// <param name="response">The HttpResponse to set the content type and status code on.</param>
+    /// <param name="headers">The headers.</param>
+    /// <param name="body">The response body.</param>
+    /// <param name="statusCode">The status code to set.</param>
+    private static void SetContentTypeAndStatusCodeV2(HttpResponse response, IDictionary<string, string>? headers, string? body, int statusCode)
+    {
+        if (headers != null && headers.TryGetValue("Content-Type", out var contentType))
+        {
+            response.ContentType = contentType;
+        }
+        else
+        {
+            response.ContentType = "text/plain; charset=utf-8"; // api gateway v2 defaults to this content type if none is provided
+        }
+
+        if (statusCode != 0)
+        {
+            response.StatusCode = statusCode;
+        }
+        // v2 tries to automatically make some assumptions if the body is valid json
+        else if (IsValidJson(body))
```

Comment:
this logic here - i havent been able to find a case where returning something from a lambda is considered _not_ json. even doing `callback(null, 1) or `callback(null, "fdasfdasfads") in the lambda, still thinks this response type is json. so im thinking maybe we should actually just check "is body present and if it is, set as json"